src/controller/data.controller.ts

```diff
@@ -0,0 +1,370 @@
+import { IData } from './types/data.interface';
+import productsData from '../lib/data/productsData.json';
+import roundNumber from '../lib/numberHelpers';
+
+interface CartItemType {
+  id: number;
+  count: number;
+}
+
+interface CartType {
+  currency: '$';
+  idArray: CartItemType[];
+}
+
+interface RangeFilters {
+  Price: number[];
+  Stock: number[];
+}
+export interface PromoCode {
+  name: string;
+  discount: number;
+  description: string;
+}
+
+interface Storage {
+  products: {
+    layout: boolean;
+    queries: string;
+    init: {
+      price: boolean;
+      count: boolean;
+    };
+    sort: {
+      price: boolean;
+      count: boolean;
+    };
+  };
+  filters: {
+    search: string;
+  };
+}
+
+export default class DataController {
```

Comment:
as we write using react we should put this logic in custon hook that process data (for the future we do not need this controller at all)

Reply:
+/-